BusinessCentral.LinterCop/Design/Rule0065CheckEventSubscriberVarKeyword.cs

```diff
@@ -0,0 +1,85 @@
+#nullable enable
+using Microsoft.Dynamics.Nav.CodeAnalysis;
+using Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using Microsoft.Dynamics.Nav.CodeAnalysis.InternalSyntax;
+
+namespace BusinessCentral.LinterCop.Design;
+
+[DiagnosticAnalyzer]
+public class Rule0065CheckEventSubscriberVarKeyword : DiagnosticAnalyzer
+{
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+        ImmutableArray.Create<DiagnosticDescriptor>(DiagnosticDescriptors.Rule0065EventSubscriberVarCheck);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.RegisterSymbolAction(CheckForEventSubscriberVar, SymbolKind.Method);
+    }
+
+    private void CheckForEventSubscriberVar(SymbolAnalysisContext context)
+    {
+        var methodSymbol = (IMethodSymbol)context.Symbol;
+        var eventSubscriberAttribute = methodSymbol.Attributes
+            .FirstOrDefault(attr => attr.AttributeKind == AttributeKind.EventSubscriber);
```

Comment:
Should this be a .SingleOrDefault instead of a .FirstOrDefault?

Reply:
SingleOrDefault iterates all elements and throws if there are multiple results based on the predicate. Not sure if this is necessary. Does the AL compiler even allow for multiple eventsubscriber attributes on one method?

Reply:
Thank foor looking into this! I tent to use a .SingleOrDefault when I only expect a single result.

But you're right, the AL compiler will throw an error if you try to add  multiple eventsubscriber attributes on one method, so unnecessary to have this extra validation.